app/Http/Controllers/EpisodeManager.php

```diff
@@ -122,4 +154,90 @@ public function uploadFileToS3(Request $request)
 
         return $s3->getDriver()->getAdapter()->getClient()->getObjectUrl('suyabay', $fileName);
     }
+
+    public function delete( Request $request )
+    {
+        $episode_id  =  $request['episode_id'];
+        $episode     = Episode::where('id', $episode_id)->delete();
+        
+        if ( $episode  === 1 ) 
+        {
+            $data = 
+            [
+                "status"    => 200,
+                "message"   => "Episode successfully deleted"
+            ];
+        }
+        
+        if ( $episode  === 0 ) 
+        {
+            $data = 
+            [
+                "status"    => 401,
+                "message"   => "episode can not be deleted"
+            ];
+        }
+
+        return $data;
+    }
+
+    public function updateEpisodeStatus ( Request $request )
+    {
+        $episode_id     =  $request['episode_id'];
+        $episode        = Episode::where('id', $episode_id)->update(['status' => 1]);
+
+        if ( $episode  === 1 ) 
+        {
+            $data =
+            [
+                "status"    => 200,
+                "message"   => "Episode successfully updated"
+            ];
+        }
+        
+        if ( $episode  === 0 ) 
+        {
+            $data = 
+            [
+                "status"    => 401,
+                "message"   => "episode can not be updated"
+            ];
+        }
+
+        return $data;
+    }
+
+    public function pendingEpisode()
+    {
+
+        $user                           = User::all();
+        $online_user                    = $user->where('active', 1)->count();
+        $offline_user                   = $user->where('active', 0)->count();
+        $numbers_of_users_on_suyabay    = $user->count();
+
+        $episodes                       = Episode::get();
+        $active_episode                 = $episodes->where('status', 1);
+        $pendding_episode               = $episodes->where('status', 0);
+
+        $data = 
```

Comment:
**Code Quality - Critical**

This method repeats  what you have in the index method. why is that?.

```php
        $user                               = User::all();
        $online_user                   = $user->where('active', 1)->count();
        $offline_user                   = $user->where('active', 0)->count();
        $numbers_of_users_on_suyabay    = $user->count();
        $episodes                       = Episode::get();
        $active_episode                 = $episodes->where('status', 1);
        $pendding_episode               = $episodes->where('status', 0);
        $channels                       = Channel::get();
        $data = 
        [
            "user" => 
                    [
                        "total"     => $numbers_of_users_on_suyabay,
                        "online"    => $online_user,
                        "offline"   => $offline_user
                    ],
            
            "episodes" =>
                    [
                        "recent"    => $episodes,
                        "active"    => $active_episode,
                        "pending"   => $pendding_episode
                    ],
            "channels" => $channels
        ];
``` 
Please refactor that and make it DRY